sdk/webpubsub/azure-messaging-webpubsub-client/src/main/java/com/azure/messaging/webpubsub/client/WebPubSubAsyncClient.java

```diff
@@ -0,0 +1,1069 @@
+                handleClientStop();
+            });
+        } else {
+            // sequenceAck task
+            if (webPubSubProtocol.isReliable()) {
+                Flux<Void> sequenceAckFlux = Flux.interval(SEQUENCE_ACK_DELAY).concatMap(ignored -> {
+                    if (clientState.get() == WebPubSubClientState.CONNECTED && session != null && session.isOpen()) {
+                        Long id = sequenceAckId.getUpdated();
+                        if (id != null) {
+                            return sendMessage(new SequenceAckMessage().setSequenceId(id));
+                        } else {
+                            return Mono.empty();
+                        }
+                    } else {
+                        return Mono.empty();
+                    }
+                });
+
+                Disposable previousTask = sequenceAckTask.getAndSet(sequenceAckFlux.subscribe());
+                if (previousTask != null) {
+                    previousTask.dispose();
+                }
+            }
+
+            // restore group
+            if (autoRestoreGroup) {
+                List<Mono<WebPubSubResult>> restoreGroupMonoList = groups.values().stream()
+                    .filter(WebPubSubGroup::isJoined)
+                    .map(group -> joinGroup(group.getName()).onErrorResume(error -> {
+                        if (error instanceof Exception) {
+                            tryEmitNext(rejoinGroupFailedEventSink,
+                                new RejoinGroupFailedEvent(group.getName(), (Exception) error));
+                        }
+                        return Mono.empty();
+                    }))
+                    .collect(Collectors.toList());
+
+                Flux.mergeSequentialDelayError(restoreGroupMonoList,
+                    Schedulers.DEFAULT_POOL_SIZE, Schedulers.DEFAULT_POOL_SIZE)
+                    .subscribeOn(Schedulers.boundedElastic()).subscribe(null, thr -> {
+                        logger.atWarning()
+                            .log("Failed to auto restore group: " + thr.getMessage());
+                    });
+            }
+        }
+    }
+
+    private void handleSessionClose(CloseReason closeReason) {
+        logger.atVerbose().addKeyValue("code", closeReason.getCloseCode()).log("Session closed");
+
+        if (clientState.get() == WebPubSubClientState.STOPPED) {
+            return;
+        }
+
+        if (isStoppedByUser.compareAndSet(true, false)
+            || clientState.get() == WebPubSubClientState.STOPPING) {
+            // send DisconnectedEvent
+            tryEmitNext(disconnectedEventSink, new DisconnectedEvent(connectionId, null));
+
+            // stopped by user
+            handleClientStop();
+        } else if (closeReason.getCloseCode() == 1008) {
+            // do not send DisconnectedEvent
+            // server likely send the DisconnectedMessage before close
+
+            // VIOLATED_POLICY
+            handleClientStop();
```

Comment:
`handleNoRecovery` for `1008`. `1008` means service don't think the connection can be recovered anymore. So client should directly start a new connection instread of try recovering